types/cli-progress/index.d.ts

```diff
@@ -144,48 +148,74 @@ export interface Preset {
     format: string;
 }
 
-export class SingleBar extends EventEmitter {
+export class GenericBar extends EventEmitter {
     /** Initialize a new Progress bar. An instance can be used multiple times! it's not required to re-create it! */
     constructor(opt: Options, preset?: Preset);
 
-    calculateETA(): void;
-    /** Force eta calculation update (long running processes) without altering the progress values. */
-    updateETA(): void;
+    /** Internal render function */
+    render(forceRendering?: boolean): void;
+
+    /** Starts the progress bar and set the total and initial value */
+    start(total: number, startValue: number, payload?: object): void;
+
+    /** Stops the progress bar and go to next line */
+    stop(): void;
 
-    formatTime(t: any, roundToMultipleOf: any): any;
+    /** Sets the current progress value and optionally the payload with values of custom tokens as a second parameter */
+    update(current: number, payload?: object): void;
+    update(payload: object): void;
 
-    getTotal(): any;
+    /** Calculate the actual progress value */
+    getProgress(): number;
 
     /** Increases the current progress value by a specified amount (default +1). Update payload optionally */
     increment(step?: number, payload?: object): void;
     increment(payload: object): void;
 
-    render(): void;
+    /** Get the total (limit) value */
+    getTotal(): number;
 
     /** Sets the total progress value while progressbar is active. Especially useful handling dynamic tasks. */
     setTotal(total: number): void;
 
-    /** Starts the progress bar and set the total and initial value */
-    start(total: number, startValue: number, payload?: object): void;
+    /** Force eta calculation update (long running processes) without altering the progress values. */
+    updateETA(): void;
+}
 
-    /** Stops the progress bar and go to next line */
-    stop(): void;
+export class SingleBar extends GenericBar {
+    /** Initialize a new Progress bar. An instance can be used multiple times! it's not required to re-create it! */
+    constructor(opt: Options, preset?: Preset);
 
-    stopTimer(): void;
+    /** Internal render function */
+    render(): void;
 
     /** Sets the current progress value and optionally the payload with values of custom tokens as a second parameter */
     update(current: number, payload?: object): void;
     update(payload: object): void;
+
+    /** Starts the progress bar and set the total and initial value */
+    start(total: number, startValue: number, payload?: object): void;
+
+    /** Stops the progress bar and go to next line */
+    stop(): void;
 }
 
 export class MultiBar extends EventEmitter {
     constructor(opt: Options, preset?: Preset);
 
+    /** add a new bar to the stack */
     create(total: number, startValue: number, payload?: any): SingleBar;
```

Comment:
this seems to could be broken if accepted, requires new optional param:
https://github.com/npkgz/cli-progress/compare/v3.9.1...v3.11.0#diff-fa0af3cdee5566f99ed64cc9c51331c4203b2fcf9e111c4412b4eac054d88c02R44

Reply:
good catch; I added the option; also found 2 missing other options